nni/compression/experiment/config/utils.py

```diff
@@ -0,0 +1,175 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+from __future__ import annotations
+from typing import Any, Tuple, Dict, List, Type
+
+from torch.nn import Module
+
+from nni.compression.pytorch.utils import count_flops_params
+from .compression import CompressionConfig
+from .vessel import CompressionVessel
+
+
+KEY_MODULE_NAME = 'module_name::'
+KEY_PRUNERS = 'pruners'
+KEY_VESSEL = '_vessel'
+KEY_ORIGINAL_TARGET = '_original_target'
+KEY_THETAS = '_thetas'
+
+
+# hard code and magic number for flops/params reward function
+def _flops_theta_helper(target: int | float | str | None, origin: int) -> Tuple[float, float]:
+    if not target or (isinstance(target, (int, float)) and target == 0):
```

Comment:
Could you add some explanations here?

Reply:
add the following docstrings,
```
the reward function is: `sigmoid(flops_retained) = 1 / (1 + exp(-theta1 * (flops_retained + theta0)))`
this helper function return a theta pair (theta0, theta1) for building a suitable (maybe) function.
the lower evaluating result (flops/params) compressed model has, the higher reward it gets.
```

![image](https://user-images.githubusercontent.com/33053116/169327547-279d14ac-af1b-4cfe-ba31-08def1f9ba38.png)